src/v_video.c

```diff
@@ -709,18 +709,110 @@ void V_DrawPatchTRTR(int x, int y, crop_t crop, patch_t *patch,
     DrawPatchInternal(x, y, crop, patch, false);
 }
 
-void V_DrawPatchFullScreen(patch_t *patch)
+//
+// V_GetPatchColor
+// Get the color of a Doom-Format
+// graphic via lumpnum
+//
+
+rgb_t V_GetPatchColor(int lumpnum)
 {
+  rgb_t col = { 0, 0, 0 };
+  int r = 0, g = 0, b = 0;
+  const byte *playpal = W_CacheLumpName("PLAYPAL", PU_STATIC);
+  int x, y, pixel_cnt = 0;
+  const byte* lump;
+  short width;
+
+  lump = W_CacheLumpNum(lumpnum, PU_STATIC);
+
+  width = *((const int16_t *) lump);
```

Comment:
Nope, no pointer type punning, please. At this point we already know that the lump is in patch format, so please use the usual `width = SHORT(patch->width)` here. Same below for `offset`, please.